src/systems/audio-zones-system.js

```diff
@@ -1,13 +1,28 @@
 // We apply the most restrictive audio parameters
 function paramsReducer(acc, curr) {
+  if (!curr) return;
   if (acc === null) acc = curr;
-  acc.gain = Math.min(acc.gain, curr.gain);
-  acc.maxDistance = Math.min(acc.maxDistance, curr.maxDistance);
-  acc.refDistance = Math.min(acc.refDistance, curr.refDistance);
-  acc.rolloffFactor = Math.max(acc.rolloffFactor, curr.rolloffFactor);
-  acc.coneInnerAngle = Math.min(acc.coneInnerAngle, curr.coneInnerAngle);
-  acc.coneOuterAngle = Math.min(acc.coneOuterAngle, curr.coneOuterAngle);
-  acc.coneOuterGain = Math.min(acc.coneOuterGain, curr.coneOuterGain);
+  if (curr.gain !== undefined) {
+    acc.gain = Math.min(acc.gain, curr.gain);
+  }
+  if (curr.maxDistance !== undefined) {
+    acc.maxDistance = Math.min(acc.maxDistance, curr.maxDistance);
+  }
+  if (curr.refDistance !== undefined) {
+    acc.refDistance = Math.min(acc.refDistance, curr.refDistance);
+  }
+  if (curr.rolloffFactor !== undefined) {
+    acc.rolloffFactor = Math.max(acc.rolloffFactor, curr.rolloffFactor);
+  }
+  if (curr.coneInnerAngle !== undefined) {
+    acc.coneInnerAngle = Math.min(acc.coneInnerAngle, curr.coneInnerAngle);
+  }
+  if (curr.coneOuterAngle !== undefined) {
+    acc.coneOuterAngle = Math.min(acc.coneOuterAngle, curr.coneOuterAngle);
+  }
+  if (curr.coneOuterGain !== undefined) {
+    acc.coneOuterGain = Math.min(acc.coneOuterGain, curr.coneOuterGain);
+  }
   return acc;
```

Comment:
Maybe this function would be simpler as a `reduce` over the possible parameter names:
```js
return ["gain", "maxDistance",...].reduce((result, [key]) => {
        if ( predicate ) {
            result[key] = value;
        }
        return result;
    }, {});
```

Reply:
yep, that's much better, thanks. Updated.